typescript-test-samples/appsync-cognito-lambda/README.md

```diff
@@ -0,0 +1,86 @@
+[![test: integration](https://img.shields.io/badge/Test-Integration-yellow)](https://img.shields.io/badge/Test-Integration-yellow)
+
+# Typescript: AWS AppSync, Amazon Cognito, AWS Lambda
+
+## Introduction
+
+This example contains introductory testing examples of TypeScript tests written for AWS AppSync with authentication provided by Amazon Cognito.
+
+The project uses [AWS CDK] for Infrastructure as Code.
+
+---
+
+## Contents
+
+-   [Introduction](#introduction)
+-   [Contents](#contents)
+-   [About this Pattern](#about-this-pattern)
+-   [About this Example](#about-this-example)
+    -   [Key Files in the Project](#key-files-in-the-project)
+-   [Run the Integration Tests](#run-the-integration-tests)
+
+---
+
+## About this Pattern
+
+### System Under Test (SUT)
+
+The SUT in this pattern is a synchronous AWS Appsync API with an AWS Lambda data source, authenticated by Amazon Cognito.
+
+![Architecture](images/architecture.png)
+
+### Goal
+
+The goal of this pattern is to provide an example of an API-level integration test for an authenticated AppSync endpoint. The pattern provides helpers for signing up a new user, getting an access token and calling AWS AppSync via an HTTP request.
+
+## [Top](#contents)
+
+## About this Example
+
+This specific sample project allows a user to execute an AppSync query to generate a random message.
+
+This project consists of an [AppSync GraphQL API](https://aws.amazon.com/appsync/), a single [AWS Lambda](https://aws.amazon.com/lambda) function, and [Amazon Cognito](https://aws.amazon.com/cognito).
+
+### Key Files in the Project
+
+-   [stack.ts](lib/stack.ts) - CDK stack definition for deployment
+-   [integration-handler.test.ts](tests/integration/integration-handler.test.ts) - API-level integration tests on a live stack
+
+## [Top](#contents)
+
+## Run the Integration Tests
+
+### Deployment
+
+[integration-handler.test.ts](tests/integration/integration-handler.test.ts)
+
+For integration tests, deploy the full stack before testing:
+
+```shell
+appsync-cognito-lambda$ npm i
+appsync-cognito-lambda$ npm run cdk deploy
+```
+
+Deploying the stack will output three variables: the AppSync URL, the Cognito User Pool ID and the Cognito Client ID. Save these values for the next step.
+
+### Environment setup
+
+The [integration tests](tests/integration/integration-handler.test.ts) need three environment variables.
+
+1. The `APPSYNC_URL` is the URL of AppSync
+2. The `COGNITO_USER_POOL_ID` is the User Pool ID of Cognito
+3. The `COGNITO_CLIENT_ID` is the Client ID of Cognito
+
+Set up the environment variables by making a copy of `env.template` and naming it `.env`. Replace the placeholder values with the values of your deployed environment (from the previous step).
+
+To run the test suite:
+
+```shell
+appsync-cognito-lambda$ npm i
+appsync-cognito-lambda$ npm run test:integration
```

Comment:
I deployed to `us-east-2`.

When I run the test, I get a region error.

```
Running one project: integration
  console.error
    signUp Error: Region is missing
at default (/Users/krygsman/code/tfc-serverless/serverless-testing-initiative/aiden-walton/serverless-test-samples/typescript-test-samples/appsync-cognito-lambda/node_modules/@smithy/config-resolver/dist-cjs/regionConfig/config.js:10:15)
        at /Users/krygsman/code/tfc-serverless/serverless-testing-initiative/aiden-walton/serverless-test-samples/typescript-test-samples/appsync-cognito-lambda/node_modules/@smithy/node-config-provider/dist-cjs/fromStatic.js:6:83
... additional stack details
```

Reply:
same error for `deleteUser`

Reply:
When I add `AWS_REGION=us-east-2` to the `.env` file, the error changes to:

`signUp CredentialsProviderError: Could not load credentials from any providers`

Note that my current terminal session has an active AWS session with Admin permissions to the account.

Reply:
Added a bit more granular logging and it looks like its the `AdminConfirmSignUpCommand` that is throwing the error.

Reply:
Re-cloned, re-built, and tests are passing now. Not sure what happened, but I suspect its my system config rather than this code. Resolving.